.github/workflows/test.yml

```diff
@@ -37,18 +37,43 @@ jobs:
     name: Build-master
     runs-on: ubuntu-latest
     steps:
+    # Create a cache for the built master image
+    - name: Restore master image cache
+      id: image_cache_master
+      uses: actions/cache@v2
+      with:
+        path: |
+          /tmp/image_cache/
+        key: ${{ github.run_id }}-image-cache-master
```

Comment:
should this be cached based on a commit id instead? Then other PRs could potentially re-use the image from the cache? Potentially an optimization we can make later.